app/routes/armoury/$character/exchange.tsx

```diff
@@ -29,25 +32,34 @@ export async function loader({ request, params }: LoaderArgs) {
 			]
 		let currentShop = await getCharacterStore(
 			auth,
-			currentCharacter?.archetype,
-			currentCharacter?.id
+			currentCharacter?.archetype!,
+			currentCharacter?.id!
```

Comment:
I'd rather check that currentCharacter exists than assert that it does. Someone going to a bad url will cause a server error rather than our own error state (which would need implementing). Or actually a redirect would be nice. If you go to a character url that doesn't exist you should get sent to /armoury